OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -384,6 +388,9 @@ protected virtual WVec CalculateMuzzleOffset(Actor self, Barrel b)
 			else
 				localOffset = localOffset.Rotate(bodyOrientation);
 
+			if (hovers != null)
+				localOffset += hovers.WorldVisualOffset;
```

Comment:
Shouldn't this be added before rotation?

Reply:
I didn't see difference in my tests, but will move it before rotation. For good measure I also added Sync to WorldVisualOffset

Reply:
I'm not sure sync is necessary.

Hover units usually don't tilt, but if they did I'm pretty sure this would cause misalignments

Reply:
I wanted it to be consistent with Turreted which _WorldOrientation_ property is based on synced _QuantizedFacings_. Not sure whether that's the correct approach either.